pymoose/moosemodule.cpp

```diff
@@ -2717,14 +2718,23 @@ int defineDestFinfos(const Cinfo * cinfo)
         // if (name.find("get") == 0 || name.find("set") == 0){
         //     continue;
         // }
-        PyGetSetDef destFieldGetSet;
+        PyGetSetDef destFieldGetSet = {.name = (char*) name.c_str()
+          , .doc= (char*) "Destination field" };
         vec.push_back(destFieldGetSet);
 
-        vec[currIndex].name = strdup(name.c_str());
-        vec[currIndex].doc = (char*) "Destination field";
+        // Dilawar:
```

Comment:
@subhacom The clang++ compiler did not compile. Because `PyGetSetDef.name` is a `const char*` therefore the function `strncpy` can't write to this field. Have a look at these changes. I don't think there is a possibility of memory leak.

See https://docs.python.org/3/c-api/structures.html#c.PyGetSetDef